yarn-project/end-to-end/src/fixtures/setup_p2p_test.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Test fixtures and utilities to set up and run a test using multiple validators
+ */
+import { type AztecNodeConfig, AztecNodeService } from '@aztec/aztec-node';
+import { type SentTx, createDebugLogger } from '@aztec/aztec.js';
+import { type AztecAddress } from '@aztec/circuits.js';
+import { type BootnodeConfig, BootstrapNode, createLibP2PPeerId } from '@aztec/p2p';
+import { type PXEService } from '@aztec/pxe';
+import { NoopTelemetryClient } from '@aztec/telemetry-client/noop';
+
+import { generatePrivateKey } from 'viem/accounts';
+
+import { getPrivateKeyFromIndex } from './utils.js';
+
+export interface NodeContext {
+  node: AztecNodeService;
+  pxeService: PXEService;
+  txs: SentTx[];
+  account: AztecAddress;
+}
+
+export function generatePeerIdPrivateKeys(numberOfPeers: number): string[] {
+  const peerIdPrivateKeys = [];
+  for (let i = 0; i < numberOfPeers; i++) {
+    peerIdPrivateKeys.push('08021220' + generatePrivateKey().substr(2, 66));
```

Comment:
Can you explain the magic number?

Reply:
Its a multiaddr prefix https://multiformats.io/multiaddr/